v1/plugins/plugins.go

```diff
@@ -862,6 +875,7 @@ func (m *Manager) Reconfigure(config *config.Config) error {
 
 	// don't erase persistence directory
 	if config.PersistenceDirectory == nil {
+		// update is ok since we have the lock
 		config.PersistenceDirectory = m.Config.PersistenceDirectory
 	}
 
```

Comment:
(this comment refers to the line below 👇, but GH won't let me comment there 😄)

Should we clone the incoming config to make absolutely sure that no one updates it outside the lock?

Reply:
Ok, I've updated this now to use a copy of the new config too.